src/engine/strat_engine/filesystem.rs

```diff
@@ -91,6 +68,11 @@ impl StratFilesystem {
     pub fn teardown(self, dm: &DM) -> EngineResult<()> {
         Ok(try!(self.thin_dev.teardown(dm)))
     }
+
+    /// Destroy the filesystem.
```

Comment:
it'll fail if it's still mounted. My take on listing preconditions was that it was for things that needed to be true for correct operation, but that weren't checked (maybe b/c checked at a higher level), and didn't return an Err, in the function itself. In this case, trying to destroy the thindev under a mounted filesystem will fail so I think that means we don't need to list it explicitly as a precondition. Thoughts?